superset-frontend/src/explore/components/controls/TextControl.jsx

```diff
@@ -55,7 +55,7 @@ export default class TextControl extends React.Component {
       if (error) {
         errors.push(error);
       } else {
-        value = parseFloat(value);
+        value = value.match(/.*(\.)$/g) ? value : parseFloat(value);
```

Comment:
maybe I'm rusty on my regex, but it seems like this would only match strings that end in a `.`. The () is also probably unnecessary since you are not using the match groups. 

maybe try?
`/[0-9]*\.[0-9]+$/g`